### ISSUE TITLE
Unhandled Error on saving entry with empty custom label name

### ISSUE THREAD

**Issue by lalitsom** [buttercup/buttercup-desktop#371]
![ggg](https://user-images.githubusercontent.com/23455413/31867456-e650df0e-b7ad-11e7-8ad2-95552955df91.JPG)

If user click on Add new field and then forget to enter label name, the save button will give this not so user friendly error message.
Maybe you could give a nice message that "Please enter label name" or maybe you can give some default label name.

**Comment by snhasani** [comment]
@sallar you can assign this to me.

I suggest we can ignore empty custom fields and we save just dirty fields. let me know what you think about this.

**Comment by perry-mitchell** [comment]
All fields require unique names, so we do require users to actually enter names before saving. Perhaps an error message is inappropriate, but saving should be prevented and the user notified somehow that they should enter the field name.

**Comment by snhasani** [comment]
@perry-mitchell I totally agree with you, but my suggest specifically is related to the custom field. Because I think its nature is optional and if user add a custom field but didn't fill it, we can just ignore it when we want submit the form. But we could have some validation rule for its value.

**Comment by perry-mitchell** [comment]
What if a user painfully types out a large amount of text and we delete it when saving because it didn’t have a label?

I would still suggest blocking saving if any part of the form is invalid. Saving while there is invalid data, even by skipping fields, is misleading I feel.

I think the whole red-highlight kind of style when submitting incomplete forms is a nice way to accomplish this, and isn’t as in-your-face as an error dialog.

**Comment by snhasani** [comment]
@perry-mitchell I did not mean we should just have a validation in submit level. I agree with you and I think we must be have validation for each field on field level when blur event.

My mean from ignoring custom field it was when user did not filed any of field inputs. We can just ignore that field when both of them (name and value) was empty.

**Comment by perry-mitchell** [comment]
> We can just ignore that field when both of them (name and value) was empty.

Ah, I understand what you meant.. Yes that definitely sounds correct to me also. I would go as far to say that we could delete that _row_ if it is entirely empty.

**Comment by snhasani** [comment]
So, based on my knowledge I must hit these:

1. Field level validation with good messaging (instant)
2. Submit level normalize and validation with good messaging

And for implement these items I think current design is not good for and we need discuss about this.

For example on field validation we don't have any space between fields row and we can not show the error message under or top of them.

And for custom fields if we want have field validation we need to show inputs with stack view.

cc @sallar @perry-mitchell

**Comment by ph1p** [comment]
I've fixed this error in my PR #400
